nni/tools/nnictl/launcher.py

```diff
@@ -88,11 +88,14 @@ def create_experiment(args):
 
     exp = Experiment(config)
     exp.url_prefix = url_prefix
-    run_mode = RunMode.Foreground if foreground else RunMode.Detach
-    exp.start(port, debug, run_mode)
 
-    _logger.info(f'To stop experiment run "nnictl stop {exp.id}" or "nnictl stop --all"')
-    _logger.info('Reference: https://nni.readthedocs.io/en/stable/reference/nnictl.html')
+    if foreground:
+        exp.run(port, debug=debug)
+
+    else:
+        exp.start(port, debug, RunMode.Detach)
+        _logger.info(f'To stop experiment run "nnictl stop {exp.id}" or "nnictl stop --all"')
+        _logger.info('Reference: https://nni.readthedocs.io/en/stable/reference/nnictl.html')
```

Comment:
these lines imply that the only different between `exp.start` and `exp.run` is that `exp.run` will wait until experiment completes, right? better to put this information in docstring.

Reply:
If I remember correctly, end users should always use `run()` according a previous discussion. `run()` has carefully designed and reviewed parameters, while `start()` has kept its "experimental" signature.
We never mention `start()` in tutorial.
I have once thought about renaming `start()` to `_start()`, but gave up due to compatibility concern.

Back to the code. I have to admit line 93 is incorrect. It should be `exp.start(..., Foreground) ; exp.wait_completion()`. I don't know why I forgot about that...